opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/view.py

```diff
@@ -15,17 +15,36 @@
 
 from fnmatch import fnmatch
 from logging import getLogger
-from typing import Optional, Set, Type
+from typing import Callable, Optional, Set, Type
 
 from opentelemetry.metrics import Instrument
 from opentelemetry.sdk.metrics._internal.aggregation import (
     Aggregation,
     DefaultAggregation,
+    _Aggregation,
+    _ExplicitBucketHistogramAggregation,
+    _ExponentialBucketHistogramAggregation,
+)
+from opentelemetry.sdk.metrics._internal.exemplar import (
+    AlignedHistogramBucketExemplarReservoir,
+    ExemplarReservoirFactory,
+    SimpleFixedSizeExemplarReservoir,
 )
 
 _logger = getLogger(__name__)
 
 
+def default_reservoir_factory(
+    aggregationType: Type[_Aggregation],
+) -> ExemplarReservoirFactory:
+    """Default reservoir factory per aggregation."""
+    if issubclass(aggregationType, _ExplicitBucketHistogramAggregation):
```

Comment:
I think if the number of buckets is 1 it should default to `SimpleFixedSizeExemplarReservoir`

Reply:
The degenerated case is actually when not `boundaries` are defined as the `AlignedHistogramBucketExemplarReservoir` has `len(boundaries) + 1` buckets - the last bucket being for outliners. As defining no boundaries is a strange idea and because it will be hard to implement in a consistent manner, I would keep the code as is.

Reply:
After discussing with @czhang771 I actually miss the point that it comes from the spec:

> * Explicit bucket histogram aggregation with more than 1 bucket SHOULD use AlignedHistogramBucketExemplarReservoir.
> * Base2 Exponential Histogram Aggregation SHOULD use a SimpleFixedSizeExemplarReservoir with a reservoir equal to the smaller of the maximum number of buckets configured on the aggregation or twenty (e.g. min(20, max_buckets)).
> * All other aggregations SHOULD use SimpleFixedSizeExemplarReservoir.

Interestingly, neither the [Java SDK](https://github.com/open-telemetry/opentelemetry-java/blob/ee7fd2744949ae0c060d28c73edfca56633dc170/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/exemplar/ExemplarReservoir.java#L89) nor the [.NET](https://github.com/open-telemetry/opentelemetry-dotnet/blob/685b73d3ac7a4d4f0b6d658650f93fb1ad166e40/src/OpenTelemetry/Metrics/MetricPoint/MetricPoint.cs#L82) implements that. They both do as we do here.

Reply:
Are you referring to the "MORE THAN" for the first bullet point?

Reply:
Indeed this is for the _more than_ case